Protocol-buffers.md

```diff
@@ -0,0 +1,125 @@
+## Tableof contents
+
+* [Introduction](#introduction)
+* [Support for protocol buffers](#support-for-protocol-buffers)
+  * [Install proto files](#install-proto-files)
+  * [Generate code from proto files](#generate-code-from-proto-files)
+* [Adding your own proto files](#adding-your-own-proto-files)
+* [Examples](#examples)
+  * [Oppia ML](#oppia-ml)
+
+## Introduction
+
+At Oppia, we need to be able to store and transfer data in a language-agnostic way, for example between our frontend and backend code. We mostly use JSON, but we sometimes use [protocol buffers](https://developers.google.com/protocol-buffers/) instead. To quote from its documentation:
+
+> Protocol buffers are a language-neutral, platform-neutral extensible mechanism for serializing structured data.
+
+Practically speaking, this means you can define a data structure in a `*.proto` file (called a proto file). Then, you can run a command to generate code for reading and writing that data structure. The magic comes from the fact that the generated code can be in many different languages. This lets you define a data structure once and use it in both Python and JavaScript, for example. We use [buf](https://docs.buf.build) to perform this code generation.
+
+Below, we'll discuss how we use protocol buffers at Oppia.
+
+## Support for protocol buffers
+
+### Install proto files
+
+We expect proto files to be defined in their own repositories, which we download and install as dependencies. In [`manifest.json`](), the `proto` section describes how to download proto files:
```

Comment:
are we expecting to add a link for `manifest.json`?

Reply:
yes; done